documentation/docs/core-concepts/anatomy-of-a-request.md

```diff
@@ -0,0 +1,82 @@
+---
+title: Anatomy of a Request
+path: /anatomy-of-a-request/
+---
+
+# Anatomy of a Request
+
+In the previous sections, we learned about Fusion.js plugins and their ability to define middleware functions that can read the `request` object and/or make changes to it. In this section, we'll cover what happens when a request comes into the Fusion.js server, the order that plugins are accessed, and when rendering occurs.
+
+We'll be using the following example for this discussion.
+
+```js
+const app = new App();
+app.register(FirstPlugin);
+app.register(SecondPlugin);
+app.register(StandalonePlugin);
+```
+
+Let's also assume that `SecondPlugin` depends on `FirstPlugin` but `StandalonePlugin` is stand-alone with no dependencies.
+
+```js
+const FirstPlugin = createPlugin({
+  middleware: () => (ctx, next) => {
+    console.log('FirstPlugin');
+    next();
+  },
+});
+
+const SecondPlugin = createPlugin({
+  // Define a dependency to FirstPlugin!
+  deps: {first: FirstPlugin},
+  middleware: () => (ctx, next) => {
+    console.log('SecondPlugin');
+    next();
+  },
+});
+
+const StandalonePlugin = createPlugin({
+  middleware: () => (ctx, next) => {
+    console.log('StandalonePlugin');
+    next();
+  },
+});
+```
+
+We've given each plugin a `middleware` definition to log its own name so we can tell when they are accessed.
+
+### On the server
+
+When Fusion.js initializes, it builds a dependency graph of all registered plugins based on what plugins depend on. This ensures that each plugin has its necessary dependencies when it needs to run.
+
+In the above example, because `SecondPlugin` depends on `FirstPlugin`, the `console.log` order that the middleware will be accessed is:
+
+1. `FirstPlugin`
+2. `StandalonePlugin`
+3. `SecondPlugin`
+
+> **Note**: The order of `FirstPlugin` and `StandalonePlugin` is not deterministic since both are not dependent on anything else. This is ok in our case though since we have no preference.
+
+### Server rendering
+
+After all plugins have run, the last task is to render out the current page on the server. This is **always** the last action that the server takes. This presents an opportunity for registered plugins to modify the HTML template if necessary to add/remove HTML elements while the request chain is processing.
+
+Finally, the server render will flush the page HTML onto the client.
+
+### On the client
+
+The client will run through the same process, stepping through each `middleware` based on the topological sort of the plugin graph.
+
+### Client rendering
+
+Lastly, the client will "render" itself. Because the page has already been written to the DOM and the user can already see content at this point, this in practice is [React hydration](https://reactjs.org/docs/react-dom.html#hydrate) which synchronizes the client state with what has already been rendered on the page.
+
+### Declaring logic after render
```

Comment:
Do we ever document examples of what you might want to do here? Maybe we should if not already.

Reply:
The only real examples we have is always about measuring request timing e.g. set a timer before, wait for the `next`, then measure the time difference. 

I linked to the plugin reference which goes into more detail and has that specific example. Ultimately though I think it is a niche case.

Reply:
I'll add some examples in-line into this article and still link out to the plugins reference docs.

Reply:
Ok, updated.